openshift/templates/iiw-book-service/iiw-book-service-deploy.overrides.sh

```diff
@@ -1,68 +1,25 @@
-# -----------------------------------------------------------------------------------
-# ===================================================================================
-printStatusMsg(){
-  (
-    _msg=${1}
-    _yellow='\033[1;33m'
-    _nc='\033[0m' # No Color
-    printf "\n${_yellow}${_msg}\n${_nc}" >&2
-  )
-}
-
-readParameter(){
-  (
-    _msg=${1}
-    _paramName=${2}
-    _defaultValue=${3}
-    _encode=${4}
-
-    _yellow='\033[1;33m'
-    _nc='\033[0m' # No Color
-    _message=$(echo -e "\n${_yellow}${_msg}\n${_nc}")
-
-    read -r -p $"${_message}" ${_paramName}
-
-    writeParameter "${_paramName}" "${_defaultValue}" "${_encode}"
-  )
-}
-
-writeParameter(){
-  (
-    _paramName=${1}
-    _defaultValue=${2}
-    _encode=${3}
-
-    if [ -z "${_encode}" ]; then
-      echo "${_paramName}=${!_paramName:-${_defaultValue}}" >> ${_overrideParamFile}
-    else
-      # The key/value pair must be contained on a single line
-      _encodedValue=$(echo -n "${!_paramName:-${_defaultValue}}"|base64 -w 0)
-      echo "${_paramName}=${_encodedValue}" >> ${_overrideParamFile}
-    fi
-  )
-}
-
-initialize(){
-  # Define the name of the override param file.
-  _scriptName=$(basename ${0%.*})
-  export _overrideParamFile=${_scriptName}.param
-
-  printStatusMsg "Initializing ${_scriptName} ..."
-
-  # Remove any previous version of the file ...
-  if [ -f ${_overrideParamFile} ]; then
-    printStatusMsg "Removing previous copy of ${_overrideParamFile} ..."
-    rm -f ${_overrideParamFile}
-  fi
-}
-
-initialize
-
-# Get the webhook URL
-readParameter "EMAIL_HOST - Please provide the host name of the email server:" EMAIL_HOST "smtp.host.io" 
+#! /bin/bash
+_includeFile=$(type -p overrides.inc)
+if [ ! -z ${_includeFile} ]; then
+  . ${_includeFile}
+else
+  _red='\033[0;31m'; _yellow='\033[1;33m'; _nc='\033[0m'; echo -e \\n"${_red}overrides.inc could not be found on the path.${_nc}\n${_yellow}Please ensure the openshift-developer-tools are installed on and registered on your path.${_nc}\n${_yellow}https://github.com/BCDevOps/openshift-developer-tools${_nc}"; exit 1;
+fi
+
+# ================================================================================================================
+# Special deployment parameters needed for injecting a user supplied settings into the deployment configuration
+# ----------------------------------------------------------------------------------------------------------------
+
+if createOperation; then
+  # Ask the user to supply the sensitive parameters ...
+  readParameter "SMTP_EMAIL_HOST - Please provide the host name of the email server:" SMTP_EMAIL_HOST "smtp.host.io"
+  readParameter "STAFF_EMAILS - Please provide any staff emails" STAFF_EMAILS "addr@mail.com"
+else
+  # Secrets are removed from the configurations during update operations ...
+  printStatusMsg "Update operation detected ...\nSkipping the prompts for WALLET_SEED secret... \n"
+  writeParameter "SMTP_EMAIL_HOST" "prompt_skipped" "false"
```

Comment:
Values for `SMTP_EMAIL_HOST` should be read back from secret storage as the variable is used to populate `ExternalNetwork` resource values.  Should be following the same pattern used with `email-verification-service-deploy.overrides.sh` in https://github.com/bcgov/indy-email-verification/pull/65

Reply:
Hi Wade not sure I follow, it is read from secret storage and from what I can tell the pattern is the same as `email-verification-service-deploy.overrides.sh` could you please ellaborate?

Reply:
woops sorry, I see what you mean